src/sentry/eventstream/snuba.py

```diff
@@ -96,7 +96,7 @@ def insert(self, group, event, is_new, is_sample, is_regression,
             logger.error('%r received unexpected tags: %r', self, unexpected_tags)
 
         self._send(project.id, 'insert', extra_data=({
-            'group_id': event.group_id,
+            'group_id': event.group_id or 0,  # group id is non nullable in Snuba.
```

Comment:
Personally, I don't care for this.

Using 0 in place of `null` is an implementation detail of how we've decided to store the data in ClickHouse. (For those not in the loop, since the ID space here for groups is `[1 … (some large number)]`, we can use `0` as a sentinel value for "no value" to avoid the extra storage cost of making the type nullable, when `0` isn't a valid ID to begin with.)

We shouldn't be exposing this to other parts of the application, in my opinion. This should end up in the event stream as `null`, and we should be translating it from `null` to `0` on the Snuba side (both in the writer, as well as in the query interface if somebody passes `['group_id', '=', 'null']` as a condition. If `0` is passed as a condition, we should probably error (or rewrite the condition to a literal `false`), since that's not a valid group ID.

Reply:
That's interesting, will hold this off and try to build the data transformation in the Snuba dataset. Let's see how it works.

Reply:
https://github.com/getsentry/snuba/pull/352

The Snuba PR that maps group_id = 0 to Null

Reply:
Now eventstream takes None as group id

Reply:
Thanks!